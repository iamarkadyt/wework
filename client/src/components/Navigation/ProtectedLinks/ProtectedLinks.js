import React from 'react'
import { NavLink } from 'react-router-dom'

const protectedLinks = ({ 
  isAuthenticated 
}) => (
  isAuthenticated 
    ? <React.Fragment>
      <NavLink to="/feed"><span>Feed</span></NavLink>
    </React.Fragment> : null
)

export default protectedLinks
