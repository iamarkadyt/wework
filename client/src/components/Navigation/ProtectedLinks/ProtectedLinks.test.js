import React from 'react'
import { shallow } from 'enzyme'
import ProtectedLinks from './ProtectedLinks'
import cloneDeep from 'lodash.clonedeep'

let mountedComponent, props

const getMockProps = () => {
  return cloneDeep({
    isAuthenticated: true
  })
}

const comp = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<ProtectedLinks {...props} />)
  }
  return mountedComponent
}

describe('ProtectedLinks', () => {
  beforeEach(() => {
    mountedComponent = undefined
    props = getMockProps()
  })

  describe('props', () => {
    describe('effects', () => {
      describe('isAuthenticated', () => {
        it('if true, everything is rendered', () => {
          props.isAuthenticated = true
          expect(comp().type()).not.toBe(null)
        })

        it('if false, nothing is rendered', () => {
          props.isAuthenticated = false
          expect(comp().type()).toBe(null)
        })
      })
    })
  })

  it('matches snapshot with isAuthenticated set to true', () => {
    props.isAuthenticated = true
    expect(comp()).toMatchSnapshot()
  })

  it('matches snapshot with isAuthenticated set to false', () => {
    props.isAuthenticated = false
    expect(comp()).toMatchSnapshot()
  })
})


