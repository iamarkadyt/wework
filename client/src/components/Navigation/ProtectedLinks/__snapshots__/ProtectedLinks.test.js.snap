// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedLinks matches snapshot with isAuthenticated set to false 1`] = `""`;

exports[`ProtectedLinks matches snapshot with isAuthenticated set to true 1`] = `
<Fragment>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/feed"
  >
    <span>
      Feed
    </span>
  </NavLink>
</Fragment>
`;
